### PR TITLE
improve: improved rating algorithm wip

### PR DESCRIPTION
**Pull request by anuraghazra**
NOTE: i suck at math, so i'm just experimenting and doing trial and error to get the correct results.

### CODE REVIEW

File: src/calculateRank.js [hunk 1]
```diff
@@ -1,3 +1,21 @@
+// https://stackoverflow.com/a/5263759/10629172
+function normalcdf(mean, sigma, to) {
```
Comment (anuraghazra) [inline src/calculateRank.js:2]: I really have no idea what this does but i'm just experimenting it's a Cumulative distribution function : https://stackoverflow.com/questions/5259421/cumulative-distribution-function-in-javascript/5263759#5263759

File: src/calculateRank.js [hunk 2]
```diff
@@ -13,12 +31,24 @@ function calculateRank({
...
-  const FOURTH_STEP = 50;
-  const FIFTH_STEP = 130;
+  const RANK_S_VALUE = 1;
+  const RANK_DOUBLE_A_VALUE = 25;
+  const RANK_A2_VALUE = 45;
+  const RANK_A3_VALUE = 60;
+  const RANK_B_VALUE = 100;
+
+  const TOTAL_VALUES =
+    RANK_S_VALUE + RANK_A2_VALUE + RANK_A3_VALUE + RANK_B_VALUE;
```
Comment (anuraghazra) [inline src/calculateRank.js:51]: Adding all the OFFSETS & conditional values and feeding it to that function, all values are result of trial and errors

File: src/calculateRank.js [hunk 3]
```diff
@@ -27,26 +57,37 @@ function calculateRank({
...
-  } else if (score > FIRST_STEP && score <= SECOND_STEP) {
-    level = "B+";
-  } else if (score > SECOND_STEP && score <= THIRD_STEP) {
-    level = "A";
-  } else if (score > THIRD_STEP && score <= FOURTH_STEP) {
-    level = "A+";
-  } else if (score > FOURTH_STEP && score <= FIFTH_STEP) {
-    level = "A++";
-  } else if (score > FIFTH_STEP) {
+  if (normalizedScore < RANK_S_VALUE) {
```
Comment (anuraghazra) [inline src/calculateRank.js:68]: normalizedScore is for some reason reverse that means the better your score is the normalized  score would be closer to 0 

Users like 
- yyx990803
- kentcdodds 
- gaearon

all are close to 0

**Comment by anuraghazra** [comment]
related to #40 #41 

https://github.com/anuraghazra/github-readme-stats/issues/41#issuecomment-658021499

@vinayaksh42 @garvit-joshi can you guys check and see if anything you want to improve or review the code?

**Comment by anuraghazra** [comment]
Development preview

**Comment by garvit-joshi** [comment]
I am not that good with maths either, but preview is quite **Amazing**

**Comment by anuraghazra** [comment]
> Development preview
> use this preview link to check the updates `https://github-readme-stats.anuraghazra1.vercel.app/api?username=kentcdodds` _(note: don't use it on production)_
> 
> 
> 
> 
> 
> 
> 
> 

I think guys this is a lot better. @garvit-joshi @vinayaksh42 perfectly balanced to A and A++ and S, S+ ranks are hard to get, even kentcdodds did not get S+ rank.

also fixed #39 as you can see MichaelDeBoey had 19k stars which was because of forks so i filtered them out.

File: src/fetchStats.js [hunk 4]
```diff
@@ -26,7 +26,7 @@ const fetcher = (variables, token) => {
           followers {
             totalCount
           }
-          repositories(first: 100, orderBy: { direction: DESC, field: STARGAZERS }) {
+          repositories(first: 100, ownerAffiliations: OWNER, isFork: false, orderBy: {direction: DESC, field: STARGAZERS}) {
```
Comment (anuraghazra) [inline src/fetchStats.js:29]: Filter out own repos, and forked repos

File: src/calculateRank.js (hunk 2 above)
Comment (SamyakJain2020) [reply src/calculateRank.js:51]: @anuraghazra Hi, I wanted to know , what are these offsets are for and how are they calculated??/

**Pull request merged**